integrations/db-mongodb/src/MongoDb.ts

```diff
@@ -0,0 +1,119 @@
+import { Collection, Db, Document, MongoClient } from 'mongodb';
+import {
+  DbItem,
+  DbPlugin,
+  DbPluginConfig,
+  HandleRequest,
+  Jovo,
+  PersistableSessionData,
+  PersistableUserData,
+  RequiredOnlyWhere,
+} from '@jovotech/framework';
+
+export interface MongoDbConfig extends DbPluginConfig {
+  /** Specify username, password and clusterUrl. Additional parameters can also be added. See https://docs.mongodb.com/drivers/node/current/fundamentals/connection/#connection-uri for more details */
+  connectionString: string;
+  /** The name of the database we want to use. If not provided, use database name from connection string. A new database is created if doesn't exist yet. */
+  databaseName?: string;
+  /** A new collection is created with that name if doesn't exist yet. */
+  collectionName?: string;
+}
+
+export type MongoDbInitConfig = RequiredOnlyWhere<MongoDbConfig, 'connectionString'>;
+
+export interface MongoDbItem {
+  id: string;
+  user?: PersistableUserData;
+  session?: PersistableSessionData;
+  createdAt?: string;
+  updatedAt?: string;
+}
+
+export class MongoDb extends DbPlugin<MongoDbConfig> {
+  /** Default name for the database */
+  readonly DEFAULT_DATABASE_NAME = 'jovo-managed-db';
+
+  /** Default name for the collection */
+  readonly DEFAULT_USERS_COLLECTION_NAME = 'users_all';
+
+  /** A connection promise to be reused as MongoDB best practice: https://docs.atlas.mongodb.com/best-practices-connecting-from-aws-lambda/#connection-examples */
+  connectionPromise: Promise<MongoClient> | undefined;
+
+  constructor(config: MongoDbInitConfig) {
+    super(config);
+  }
+
+  getDefaultConfig(): MongoDbConfig {
+    return {
+      ...super.getDefaultConfig(),
+      connectionString: '<YOUR-MONGODB-URI>',
+    };
+  }
+
+  mount(parent: HandleRequest): Promise<void> | void {
+    super.mount(parent);
+  }
+
+  async initialize(): Promise<void> {
+    this.config.databaseName = this.config.databaseName;
```

Comment:
I think lines 58 and 58 can be removed.

Reply:
Oops. I would like the connectionPromise to be accesible from any component.
How could it be done?